### ISSUE TITLE
workflow never finish

### ISSUE THREAD

**Issue by xzhub** [reanahub/reana-demo-helloworld#26]
Hi,
  I am testing reana with minicube, however, the hello world demo is never finished, it always tells me it is running:

```
$ reana-client workflow status
NAME       RUN_NUMBER   ID                                     USER                                   ORGANIZATION   STATUS
workflow   6            cd6709d6-cc3e-4ec3-aac6-fead165cd7bb   00000000-0000-0000-0000-000000000000   default        running


$ kubectl get pods
NAME                                     READY     STATUS    RESTARTS   AGE
cwl-default-worker-6994f848b-kx8qt       1/1       Running   4          7h
job-controller-7d645b56f4-sztbl          1/1       Running   1          7h
message-broker-5d7f64d7cf-r66ff          1/1       Running   1          7h
server-686fc76f9c-h9gz7                  1/1       Running   1          7h
wdb-777d5b6756-tf5cp                     1/1       Running   1          7h
workflow-controller-5f6dcc799f-5b6fs     1/1       Running   1          7h
workflow-monitor-674b6c7cc5-6fxxj        1/1       Running   1          7h
yadage-alice-worker-c5b667855-vtrbb      1/1       Running   1          7h
yadage-atlas-worker-7996b4ffd8-7jjlh     1/1       Running   1          7h
yadage-cms-worker-78d5789b74-pg4pl       1/1       Running   1          7h
yadage-default-worker-5444dc95d6-tzdnr   1/1       Running   1          7h
yadage-lhcb-worker-5b79bff798-zfg86      1/1       Running   1          7h
zeromq-msg-proxy-95c55c6f-rz9xh          1/1       Running   1          7h
```

what should I do for troubleshooting?

**Comment by xzhub** [comment]
I've tried minikube v0.23.0 and v0.28.2; kubectl 1.8, 1.9.4 etc; and reana-cluster 0.2.0, 0.3.0 and latest yaml file. Seems none of them works. Could any of you tell me a tested working version that I can run this example? Thanks.

**Comment by xzhub** [comment]
Use minikube v0.23 to start a v1.6.4 k8s env, then use python 2.7 virtualenv to install reana-cluster fixed the issue. Previous I was using v1.94 k8s env and python 3. 

Here is the log

```
#install minikube/kubectl

curl -Lo minikube https://storage.googleapis.com/minikube/releases/v0.23.0/minikube-linux-amd64
chmod +x minikube
sudo mv minikube  /usr/local/bin/

wget https://storage.googleapis.com/kubernetes-release/release/v1.9.4/bin/linux/amd64/kubectl
chmod +x kubectl
sudo mv kubectl /usr/local/bin/

#start k8s v1.6.4 (this is important)
minikube start --kubernetes-version=“v1.6.4”

#install reana cluster (make sure to use python2)
virtualenv p2-reana-cluster -p python27
source p2-reana-cluster/bin/activate 
pip install reana-cluster
pip install reana-client

#start the reana cluster
reana-cluster init
reana-cluster status

#run one analysis
git clone git@github.com:reanahub/reana-demo-helloworld.git
cd reana-demo-helloworld/

eval $(reana-cluster env)
export REANA_WORKON=$(reana-client workflow create -f reana-cwl.yaml)

reana-client code upload ./code/helloworld.py
reana-client inputs upload ./inputs/names.txt
reana-client workflow start
reana-client workflow status
reana-client outputs list

reana-client outputs download greetings.txt
File greetings.txt downloaded to ./outputs/

$ cat ./outputs/greetings.txt
Hello John Doe!
Hello Jane Doe!
```

**Comment by tiborsimko** [comment]
@xzhub Indeed REANA v0.2.0 requires an older version of minikube/kubectl tools (0.23/1.6), while the REANA `master`branch -- i.e. the bleeding-edge version that will be released as v0.3.0 later once we stabilise the new APIs  -- will run with basically current versions of minikube/kubectl tools (0.28/1.10).
